### PR TITLE
[LMS-74][INTEGRATION] Search function

### PR DESCRIPTION
**Pull request by kdacudag**
## Issue Link
https://framgiaph.backlog.com/view/LMS-74

## Defintion of Done

- Be able to see searchbar
- Be able to type to searchbar
- Be able to change URL to include search query
- Be able to debounce search for 500ms
- Be able to interact with API
- Be able to display search results
- Be able to display error message for no results
- Be able to still click the grid or list icon buttons

## Steps to reproduce
- Install packages for FE and BE first
- Go to http://localhost:3000/trainer/courses/list
- Enter search keywords insde searchbar field
## Affected Components / Functionalities / Page
- SearchBar reusable component
- View as component
- Courses serializers

## Test Cases
_will update soon..._

## Screenshots
chrome_xDEEeECIYZ

### CODE REVIEW

**Review by ragg09-sunasterisk — COMMENTED** [review]
I encounter an error while testing the functionality, 
search_issue

File: client/src/pages/trainer/courses/list/index.tsx [hunk 1]
```diff
@@ -1,18 +1,37 @@
...
+import debounce from '@/src/utils/debounce';
+import { useRouter } from 'next/router';
+import axiosInstance from '@/src/apis';
 
 const View = (): ReactNode => {
   const [selectedView, setSelectedView] = useState('grid');
   const [gridIconColor, setGridIconColor] = useState('stroke-blue-500');
   const [listIconColor, setListIconColor] = useState('');
-  const listOfCourses = Array(5).fill('Course Title');
+  // const listOfCourses = Array(5).fill('Course Title');
```
Comment (ragg09-sunasterisk) [inline client/src/pages/trainer/courses/list/index.tsx:17]: remove comment

File: api/app_sph_lms/api/serializers.py [hunk 2]
```diff
@@ -9,6 +9,7 @@ class Meta:
         
 class CourseSerializer(serializers.ModelSerializer):
     category_id = CourseCategorySerializer(many=True, read_only=True)
+    category_name = serializers.CharField(source='course_category.name', read_only=True)
     class Meta:
         model = Course
         fields = "__all__"
```
Comment (ragg09-sunasterisk) [inline api/app_sph_lms/api/serializers.py:15]: add new line

**Review by Marielleregner7 — APPROVED** [review]
**APPROVED**
- [x] Search bar is visible and interactive.
- [x] URL changes based on the search query.
- [x] displays accurate search results.
- [x] displays error message for no results.
- [x] quick response.
- [ ] still click the grid or list icon buttons.

Comments: 4
Test time: 45 mins
Iterations: 1

Proof:
searchkurt

**Comment by ragg09-sunasterisk** [comment]
Comments: 4
Iterations: 1

**Pull request merged**